### PR TITLE
Migrate from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by albertosottile**
In this PR, the CI flow that was on Travis CI is migrated to GitHub Actions. Some minor changes to the build chain and/or to the codebase were also applied, which are further detailed platform by platform hereafter.

- **Windows/AppVeyor**: no changes for now. As Actions also support Windows, we might migrate this part in the future.
- **AppImage**: no changes needed to the workflow.
- **Debian**: almost no changes needed, a call to `apt-get update` was added in the test script.
- **Snapcraft**
  - all the snaps built did not work in field tests. Hence, we decided to stop building snaps in our CI/CD. It remains to be seen what we will do with the version that is currently on the Snap store. 
  - ~~the `snapcraft.yaml` recipe was completely rewritten to use `core18` as base snap, Python 3.6, and PySide2. After this change, we will not have a released platform based on Qt 4 anymore.~~
  - ~~this could also allow us to increase the minimum Python version for Syncplay to 3.6. On paper, we still support `>=3.4`, hence I reverted in this PR the f-string accidentally introduced in #327. However, `python_mpv_jsonipc` [already requires 3.6](https://www.google.com/url?q=https://github.com/iwalton3/python-mpv-jsonipc/blob/5a877b42b857d810face2b095d457b8a977cafe7/setup.py%23L23&sa=D&source=hangouts&ust=1609246315009000&usg=AFQjCNH-aGygbJnUTbJtD6SEmycL9YkjTg).~~
- **macOS**
  - the build mechanism was completely changed since Travis. Before, we used Homebrew to install Python, Qt, and PySide2. Now we use the version of Python provided by the build image and we install PySide2 from the standalone wheels (which include an embedded copy of Qt).
  - Pros: we do not depend on Homebrew anymore (and on when they upgrade their formulas/bottles). We can build on macOS 10.15 (the image provided by GitHub Actions) a package that works also on older systems (tested down to 10.13).
  - Cons: a few bugs on PySide2 require some manual steps that make the build flow brittle (see [here](https://bugreports.qt.io/browse/PYSIDE-1458) and [here](https://bugreports.qt.io/browse/PYSIDE-1459)). An extra symlink creation is required, it Is unclear if this is a bug on py2app or a misconfiguration of the macOS image. **EDIT**: I created an [issue](https://github.com/ronaldoussoren/py2app/issues/325) on py2app.
  - Known issues: for the moment, we cannot use versions newer than Python 3.7/PySide2 5.13.1 due to a limitation of py2app (see [here](https://github.com/ronaldoussoren/py2app/issues/271)).

A few general comments about the two CI platforms:
- **Travis**: the `.travis.yml` file was deleted. The helper folder `travis` is now named `ci`. The build status badge in README now points to GitHub Actions.
- **GitHub Actions**: 
  - artifacts are now stored directly on GitHub (e.g. [here](https://github.com/Syncplay/syncplay/actions/runs/449065718)) instead of Bintray so, no extra authentication is needed. The Bintray helper files were deleted. AppVeyor still deploys on BIntray but does not require these helper files.
  - artifacts are zipped automatically before being downloaded from the UI. This is a [known limitation](https://github.com/actions/upload-artifact/issues/39) of GitHub Actions that is planned to be removed.

### CODE REVIEW

**Review by daniel-123 — COMMENTED** [review]
I don't see any blocking problems with this PR.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      - name: Deploy
+        uses: actions/upload-artifact@v2
+        with:
+          name: syncplay_${{ env.VER }}_amd64.snap
+          path: |
+            dist_actions/syncplay_${{ env.VER }}_amd64.snap
+
+  appimage:
+    name: Build AppImage
+    runs-on: ubuntu-16.04
```
[the remark this answers is not in the document]
Comment (albertosottile) [reply .github/workflows/build.yml:101]: Not really, I just copied what we had on Travis. I just tried this on my fork and the built AppImage works fine. I will change this image to `ubuntu-latest` in the PR. 

Side note from the test build:
>  Ubuntu-latest workflows will use Ubuntu-20.04 soon. For more details, see https://github.com/actions/virtual-environments/issues/1816

File: snapcraft.yaml [hunk 2]
```diff
@@ -10,28 +10,55 @@ description: |
...
 grade: stable
-
-parts:
-  syncplay:
-    plugin: python
-    source: .
-    stage-packages:
-      - python3-pyside
-    after: [desktop-qt4]
+base: core18
```
[the remark this answers is not in the document]
Comment (albertosottile) [reply snapcraft.yaml:19]: To be honest, I am not even entirely sure that it worked with `core18`, as we have troubles in testing the built snap... Long story short, the recipe is already extremely cumbersome and I am not very confident that all its parts are needed. I spent quite some time changing line by line before I could build something at least usable.

Anyway, I tried to just replace the base with `core20` and stuff started to break again. I could start over but for the moment I honestly do not see the value of such an effort...

[the remark this answers is not in the document]
Comment (daniel-123) [reply snapcraft.yaml:19]: Well, given that link you provided outright lays out the sorry state of python + core20 I think it's perfectly reasonable to keep it as is.

**Pull request merged**